packages/embark/src/lib/core/plugin.js

```diff
@@ -261,11 +261,15 @@ Plugin.prototype.registerImportFile = function(importName, importLocation) {
   this.addPluginType('imports');
 };
 
-Plugin.prototype.registerActionForEvent = function(eventName, cb) {
+Plugin.prototype.registerActionForEvent = function(eventName, options, cb) {
+  if (typeof options === 'function') {
+    cb = options;
+    options = {};
+  }
   if (!this.eventActions[eventName]) {
     this.eventActions[eventName] = [];
   }
-  this.eventActions[eventName].push(cb);
+  this.eventActions[eventName].push({action: cb, options: Object.assign({priority: 50}, options)});
```

Comment:
I know we've discussed this offline, but I think it's worth putting `50` in a const with some semantic meaning. Like `DEFAULT_PRIORITY` or something like that, so other people that run into that code have a sense what this number means.